Documentation/guides/port_arm_cm4.rst

```diff
@@ -0,0 +1,935 @@
+|                 | IDLE task stack       |                                                                                                                                             |
++                 +-----------------------+                                                                                                                                             +
+|                 | INTC                  |                                                                                                                                             |
++                 +-----------------------+                                                                                                                                             +
+|                 | Timer                 |                                                                                                                                             |
++-----------------+-----------------------+--------------------------------------------------------------------------------------------------------------------+------------------------+
+| Board directory | board_app_initialize  | see :doc:`/applications/nsh/customizing`                                                                           |see :ref:`out-of-tree`. |
++                 +-----------------------+--------------------------------------------------------------------------------------------------------------------+                        |
+|                 |<arch>_board_initialize| see : BOARD_EARLY_INITIALIZ or BOARD_LATE_INITIALIZE of https://github.com/apache/nuttx/blob/master/sched/Kconfig. |                        |
++-----------------+-----------------------+--------------------------------------------------------------------------------------------------------------------+------------------------+
+
+If got the pass of Linking, you would see follwoing result.
+
+.. code-block::
+
+  $ make
+  Create version.h
+  LN: platform/board to /home/lassdk/workdir/nuttx2/apps/platform/dummy
+  Register: nsh
+  Register: sh
+  CPP:  /home/lassdk/workdir/nuttx2/nuttx/../boards/cb2sdb2/scripts/cb2sdb2.ld-> /home/lassdk/workdir/nuttx2/nuttx/../boarLD: nuttx
+
+Step7 : Verifying
+-----------------
+
+To execulte and debug on the evalutaion board, add following configurations.
+
+.. code-block::
+
+  CONFIG_RAW_BINARY=y
+  CONFIG_DEBUG_SYMBOLS=y
+
+If your implementations were good, you would see following logs.
+
+.. code-block::
+
+  NuttShell (NSH) NuttX-12.4.0
+  nsh>
+  nsh>
+  nsh>
+
+After comfirming to boot NuttShell(NSH), try to execute "apps/testing/ostest".
+The "apps/testing/ostest" needs following configurations.
+
+.. code-block::
+
+  CONFIG_BUILTIN=y
+  CONFIG_NSH_BUILTIN_APPS=y
+  CONFIG_SCHED_WAITPID=y
+  CONFIG_TESTING_OSTEST=y
+
+And one more test is recomended to check the timer implementation whether the kernel could count the time accurately or not.
```

Comment:
Don't think this should be in the scope of this guide.
Previously the goal was to have NuttShell show up